www/widgets/GenWidget.js

```diff
@@ -0,0 +1,90 @@
+/* global Widget, NNE */
+/*
+  -----------
+     info
+  -----------
+
+  This Code Generator Widget extends the base Widget class to add methods
+  for more easily creating CSS generator Widgets.
+
+  -----------
+     usage
+  -----------
+
+  // this class inherits all the properties/methods of the base Widget class
+  // refer to www/js/Widget.js to see what those are
+  // or take a look at the wiki...
+  // https://github.com/netizenorg/netnet.studio/wiki/Creating-Widgets
+
+  this.codeField = this.createCodeField({
+    value: 'font-size 24px;',
+    change: (e) => this._codeFieldUpdate(e)
+  })
+
+  this.slider = w.createSlider({
+    background: '#f00',
+    min: 0,
+    max: 100,
+    label: 'PX'
+    change: (e) => this._fontSliderUpdate(e)
+  })
+
+*/
+class GenWidget extends Widget {
+  constructor (opts) {
+    super(opts)
+    console.log('genwidget constructed')
+  }
+
+  createCodeField (opts) {
+    const el = document.createElement('div')
+    const input = document.createElement('input')
+    input.setAttribute('type', 'text')
+    input.addEventListener('change', opts.change())
+
+    const insertButton = document.createElement('button')
+    insertButton.setAttribute('type', 'button')
+    insertButton.appendChild(document.createTextNode('insert'))
+    insertButton.addEventListener('click', insert())
+    function insert () {
+      NNE.cm.replaceSelection(input.value)
+    }
+
+    el.appendChild(input)
+    el.appendChild(insertButton)
+    return el
+  }
+
+  createSlider (opts) {
+    const el = document.createElement('div')
+    el.className('gen-slider-div')
+    const slider = document.createElement('input')
+    slider.setAttribute('type', 'range')
+    slider.setAttribute('min', opts.min || '1')
+    slider.setAttribute('max', opts.max || '100')
+    slider.setAttribute('step', opts.step || '1')
+    slider.setAttribute('value', opts.value || '50')
+    slider.className = 'gen-slider'
+    slider.addEventListener('change', opts.change())
```

Comment:
i'm guessing this is why e is undefined, what u're doing is executing the `opts.change()` function && whatever that function happens to return is what u're passing in as the callback for `'change'`. what u've got here is essentially the same as:
```js
const val = opts.change() 
slider.addEventListener('change', val)
```

what u want to be passing in as the callback for `'change'` is a function (not what that functions returns), which can be done a few diff ways:
```js
function callback (e) {
  opts.change(e)
}
slider.addEventListener('change', callback)
```
or
```js
slider.addEventListener('change', function (e) {
  opts.change(e)
})
```
or
```js
slider.addEventListener('change', (e) => {
  opts.change(e)
})
```
or
```js
slider.addEventListener('change', (e) => opts.change(e))
```

Reply:
got it, im changing it to the last one. also.. just wondering then why wouldnt this work?
`slider.addEventListener('change', opts.change)`